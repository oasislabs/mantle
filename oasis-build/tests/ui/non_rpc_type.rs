use oasis_std::Context;

#[derive(oasis_std::Service, Default)]
pub struct Counter(usize);

impl Counter {
    pub fn new(ctx: &Context, start_count: Box<u32>) -> Result<Self, ()> {
        unimplemented!();
    }
}

fn main() {
    oasis_std::service!(Counter);
}
