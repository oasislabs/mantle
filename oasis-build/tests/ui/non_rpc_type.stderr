error: `alloc::boxed::Box` cannot be converted to an RPC type
 --> $DIR/non_rpc_type.rs:7:5
  |
7 | /     pub fn new(ctx: &Context, start_count: Box<u32>) -> Result<Self, ()> {
8 | |         unimplemented!();
9 | |     }
  | |_____^

error: aborting due to previous error

